src/os/unix/pa_pthread_util.c

```diff
@@ -0,0 +1,116 @@
+ * Copyright (c) 1999-2024 Ross Bencina, Phil Burk
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * The text above constitutes the entire PortAudio license; however,
+ * the PortAudio community also makes the following non-binding requests:
+ *
+ * Any person wishing to distribute modifications to the Software is
+ * requested to send the modifications to the original developer so that
+ * they can be incorporated into the canonical version. It is also
+ * requested that these non-binding requests be included along with the
+ * license above.
+ */
+
+#if !PAUTIL_USE_POSIX_ADVANCED_REALTIME && (defined(WIN32) || defined(_WIN32))
+#include <windows.h>
+#endif
+
+#include "pa_pthread_util.h"
+#include "pa_debugprint.h"
+
+PaUtilClockId PaPthreadUtil_NegotiateCondAttrClock( pthread_condattr_t *cattr )
+{
+#if PAUTIL_USE_POSIX_ADVANCED_REALTIME
+    /* Set most suitable timeout clock and return its id.
+       If a clock can't be set, return the default clock.
+    */
+    clockid_t clockId;
+
+/* try each potential clockid in order of preferences until one succeeds:*/
+#if defined(CLOCK_BOOTTIME )
+    if( pthread_condattr_setclock( cattr, CLOCK_BOOTTIME ) == 0 )
+        return CLOCK_BOOTTIME;
+#endif
+
+#if defined(CLOCK_MONOTONIC)
+    if( pthread_condattr_setclock( cattr, CLOCK_MONOTONIC ) == 0 )
+        return CLOCK_MONOTONIC;
+#endif
+
+#if defined(CLOCK_REALTIME)
+    if( pthread_condattr_setclock( cattr, CLOCK_REALTIME ) == 0 )
+        return CLOCK_REALTIME;
+#endif
+
+    /* fallback to returning the current clock id*/
+    if ( pthread_condattr_getclock( cattr, &clockId) == 0 )
+        return clockId;
+
+    /* fallback to returning the default expected clock id*/
+    PA_DEBUG(( "%s: could not configure condattr clock\n", __FUNCTION__));
+    return CLOCK_REALTIME;
+#else /* not PAUTIL_USE_POSIX_ADVANCED_REALTIME */
+    return 0; /* dummy value */
```

Comment:
Coincidently, CLOCK_REALTIME is defined as 0.

Reply:
> Coincidently, CLOCK_REALTIME is defined as 0.

It doesn't matter but I'm not sure whether that's true on all platforms.